database/csv/csv.go

```diff
@@ -0,0 +1,132 @@
+package csv
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type CSVDatabase struct {
+	csv_files  sync.Map
+	maxWorkers int
+
+	msgs chan interface{}
+
+	ctx      context.Context
+	cancel   context.CancelFunc
+	stopOnce sync.Once
+	g        *errgroup.Group
+}
+
+func NewCSVDatabase(ctx context.Context, filePaths map[string]string, maxWorkers int) (*CSVDatabase, error) {
+	if len(filePaths) == 0 {
+		return nil, fmt.Errorf("no file paths provided")
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	g, ctx := errgroup.WithContext(ctx)
+	db := &CSVDatabase{
+		maxWorkers: maxWorkers,
+		ctx:        ctx,
+		cancel:     cancel,
+		g:          g,
+	}
+	for name, path := range filePaths {
+		csvFile, err := newCSVFile(path)
+		if err != nil {
+			return nil, err
+		}
+		db.csv_files.Store(name, csvFile)
+	}
+
+	// Start workers
+	for i := 0; i < maxWorkers; i++ {
+		g.Go(db.csvWorker)
+	}
```

Comment:
Given the workers start in the instantiation. I'm assuming we'll interface with this by sending records to `db.msgs`? (As opposed to calling some function on db and passing in records) This seems to make sense for handling any new requests concurrently

(Mostly just comprehension check for myself)

Reply:
Yup exactly, I defined a `SendMsg` func [here](https://github.com/IlliniBlockchain/etl-bitcoin/pull/14/files#diff-66034a274276421c4c95008dc3eea5bc9fb472af7c76403afce8c9be501c1bb8R66) which the workers listen to [here](https://github.com/IlliniBlockchain/etl-bitcoin/pull/14/files#diff-66034a274276421c4c95008dc3eea5bc9fb472af7c76403afce8c9be501c1bb8R71)